### ISSUE TITLE
fix: Clicking on intermediate state should go to checked in FluentCheckbox

### ISSUE THREAD

**Issue by JamesNK** [microsoft/fluentui-blazor#1096]
# 🐛 Bug Report

FluentCheckbox now has an intermediate state. Clicking on the intermediate state goes to a false state. This feels different to other checkboxes I've used like this. I think clicking on an intermediate state should go to a true state. e.g. click intermediate checkbox -> checked checkbox.

Discussion about this - https://ux.stackexchange.com/questions/136581/semicheck-or-indeterminate-checkbox-behaviour. Highest rated reply recommends going to a checked state.

Examples:
* https://demos.telerik.com/aspnet-ajax/treeview/examples/functionality/checkboxes/defaultcs.aspx
* https://ej2.syncfusion.com/demos/treeview/check-box/
* https://demos.componentone.com/ASPNET/MVCExplorer/TreeView/Checkboxes

## 💻 Repro or Code Sample

```xml
<FluentCheckbox Label="All"
                ThreeState="true"
                ShowIndeterminate="false"
                @bind-CheckState="Model.AreAllTypesVisible" />
```

## 🤔 Expected Behavior

Click intermediate checkbox -> checked checkbox.

## 😯 Current Behavior

Click intermediate checkbox -> unchecked checkbox.

## 🌍 Your Environment

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Microsoft Edge, Google Chrome, Apple Safari, Mozilla FireFox]
* .NET and FAST Version [e.g. 1.8.0]

**Comment by dvoituron** [comment]
Hi,
If I understand your request correctly, you would like the order (in ThreeState=True) to be `[ ] -> [-] -> [x] -> [ ]`?
Indeed, at present the order is `[ ] -> [x] -> `-` -> [ ]`, as in Windows (WPF et WinForms).

Examples:
- WinForms
   ![ThreeState-WinForm](https://github.com/microsoft/fluentui-blazor/assets/8350694/78bbef07-edd1-4189-8a46-433d2d34b096)

- WPF
   ![ThreeState-WPF](https://github.com/microsoft/fluentui-blazor/assets/8350694/475bc15d-8917-4c73-b4d8-d2f34ced76a3)

The only example I've found that doesn't respect this order is mudBlazor.
![ThreeState-MudBlazor](https://github.com/microsoft/fluentui-blazor/assets/8350694/1ea41236-8cf3-4c15-8891-70a49e5a1e58)

All the examples you show are linked to a TreeView where the behavior might be different in this case.
We could add a global option to define this order. Let's have a look

**Comment by dvoituron** [comment]
Fixed in #1099
